samples/HealthChecksSample/GCInfoHealthCheck.cs

```diff
@@ -2,21 +2,58 @@
 using System.Collections.Generic;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Options;
 
 namespace HealthChecksSample
 {
     // This is an example of a custom health check that implements IHealthCheck.
     //
-    // See CustomWriterStartup to see how this is registered.
+    // This example also shows a technique for authoring a health check that needs to be registered
+    // with additional configuration data. This technique works via named options, and is useful
+    // for authoring health checks that can be disctributed as libraries.
+    
+    public static class GCInfoHealthCheckBuilderExtensions
+    {
+        public static IHealthChecksBuilder AddGCInfoCheck(
+            this IHealthChecksBuilder builder, 
+            string name, 
+            HealthStatus? failureStatus = null, 
+            IEnumerable<string> tags = null, 
+            long? thresholdInBytes = null)
+        {
```

Comment:
I'm tempting to put some kind of `Action<HealthCheckRegistration>` in here, but it doesn't play super nice with all of these signatures.